ctapipe_io_lst/calibration.py

```diff
@@ -574,6 +581,48 @@ def interpolate_pseudo_pulses_data_from_20181010_to_20191104(waveform, fc, fc_ol
                             interpolate_spike_A(waveform, gain, spike_A_position, pix)
         return waveform
 
+    def spike_corr(self, event, tel_id):
+        """
+        Subtract corresponding spike ADC counts
+
+        Parameters
+        ---------
+        event : `ctapipe` event-container
+        tel_id : id of the telescope
+        """
+
+        lst = event.lst.tel[tel_id]
+
+        # If R1 container exists, update it inplace
+        if isinstance(event.r1.tel[tel_id].waveform, np.ndarray):
+            container = event.r1.tel[tel_id]
+        else:
+            # Modify R0 container. This is to create pedestal files.
+            container = event.r0.tel[tel_id]
+
+        waveform = container.waveform.copy()
+
+        
+        # We have 2 functions: one for data from 2018/10/10 to 2019/11/04 and
+        # one for data from 2019/11/05 (from Run 1574) after update firmware.
+        # The old readout (before 2019/11/05) is shifted by 1 cell.
+        run_id = event.lst.tel[tel_id].svc.configuration_id        
+        if run_id > LAST_RUN_WITH_OLD_FIRMWARE:
+            subtract_spikes = subtract_spikes_jit
+        else:
+            subtract_spikes = subtract_spikes_jit_data_from_20181010_to_20191104
+
+        # corresponding spike adc counts are hard-coded for now. Those values will be loaded from a pedestal file.
+        spike_adc = np.array([47, 54, 8])
```

Comment:
those hard coded values are indeed not very clear, why not implement reading them from the pedestal file immediatelly? 

also, do you have enough statistics in the general pedestal file to estimate those values, or is a special run needed?
if a special run is needed, and no night-to-night changes are expected, another possibility would be to pass those values via configuration file (with those values as defaults)

Reply:
See https://github.com/cta-observatory/cta-lstchain/pull/725

the standard 20k Events seems enough if we only need spike heights per pixel and not per capacitor.